app-android/src/main/kotlin/com/ustadmobile/port/android/view/RoleListFragment.kt

```diff
@@ -0,0 +1,136 @@
+
+
+    class RoleListViewHolder(val itemBinding: ItemRoleListItemBinding)
+        : RecyclerView.ViewHolder(itemBinding.root)
+
+    class RoleListRecyclerAdapter(var presenter: RoleListPresenter?)
+        : SelectablePagedListAdapter<Role, RoleListViewHolder>(DIFF_CALLBACK) {
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
+                : RoleListViewHolder {
+
+            val itemBinding = ItemRoleListItemBinding.inflate(
+                    LayoutInflater.from(parent.context), parent, false)
+            itemBinding.presenter = presenter
+            return RoleListViewHolder(itemBinding)
+        }
+
+        override fun onBindViewHolder(holder: RoleListViewHolder, position: Int) {
+            val item = getItem(position)
+            holder.itemBinding.role = item
+            holder.itemBinding.presenter = presenter
+            holder.itemView.setSelectedIfInList(item, selectedItems, DIFF_CALLBACK)
+        }
+
+        override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
+            super.onDetachedFromRecyclerView(recyclerView)
+            presenter = null
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val view = super.onCreateView(inflater, container, savedInstanceState)
+        dbRepo = UmAccountManager.getRepositoryForActiveAccount(requireContext())
+        mPresenter = RoleListPresenter(requireContext(), UMAndroidUtil.bundleToMap(arguments),
+                this, this, UstadMobileSystemImpl.instance,
+                UmAccountManager.getActiveDatabase(requireContext()),
+                UmAccountManager.getRepositoryForActiveAccount(requireContext()),
+                UmAccountManager.activeAccountLiveData)
+        mDataBinding?.presenter = mPresenter
+        mDataBinding?.onSortSelected = this
+        mNewItemRecyclerViewAdapter = NewItemRecyclerViewAdapter(this,
+                requireContext().getString(R.string.create_new,
+                        requireContext().getString(R.string.role)))
+        mDataRecyclerViewAdapter = RoleListRecyclerAdapter(mPresenter)
+        mPresenter?.onCreate(savedInstanceState.toStringMap())
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mActivityWithFab?.activityFloatingActionButton?.text =
+                requireContext().getString(R.string.role)
+    }
+
+    override fun onClick(view: View?) {
+        //TODOne: Uncomment when Edit stuff in there
+        activity?.prepareRoleEditCall {
+            if(it != null) {
+                finishWithResult(it)
+            }
+        }?.launchRoleEdit(null)
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mPresenter = null
+        dbRepo = null
+    }
+
+
+    override fun onMessageIdOptionSelected(view: AdapterView<*>?,
```

Comment:
This function can be removed. It is already handled by the the parent class (UstadListViewFragment)